mono/metadata/cominterop.c

```diff
@@ -1752,31 +1737,17 @@ ves_icall_System_Runtime_InteropServices_Marshal_GetComSlotForMethodInfoInternal
 }
 
 /* Only used for COM RCWs */
-MonoObject *
-ves_icall_System_ComObject_CreateRCW (MonoReflectionType *type)
+MonoObjectHandle
+ves_icall_System_ComObject_CreateRCW (MonoReflectionTypeHandle ref_type, MonoError *error)
 {
-	ERROR_DECL (error);
-	MonoClass *klass;
-	MonoDomain *domain;
-	MonoObject *obj;
-	
-	domain = mono_object_domain (type);
-	klass = mono_class_from_mono_type (type->type);
-
-	/* call mono_object_new_alloc_specific_checked instead of mono_object_new
-	 * because we want to actually create object. mono_object_new checks
-	 * to see if type is import and creates transparent proxy. this method
-	 * is called by the corresponding real proxy to create the real RCW.
-	 * Constructor does not need to be called. Will be called later.
-	*/
-	MonoVTable *vtable = mono_class_vtable_checked (domain, klass, error);
-	if (mono_error_set_pending_exception (error))
-		return NULL;
-	obj = mono_object_new_alloc_specific_checked (vtable, error);
-	if (mono_error_set_pending_exception (error))
-		return NULL;
+	MonoDomain * const domain = MONO_HANDLE_DOMAIN (ref_type);
+	MonoType * const type = MONO_HANDLE_GETVAL (ref_type, type);
+	MonoClass * const klass = mono_class_from_mono_type (type);
 
-	return obj;
+	// mono_object_new_handle => mono_object_new_by_vtable handle
```

Comment:
You completely ignored the comment that explicitly warns against using `mono_object_new` directly. Is this change warranted and correct?

Reply:
No I didn't but I agree it is confusing.
When we converted to coop creating functions we updated the names/meanings.
`mono_object_new` is deprecated and external-only/mostly.
There is no significant use for this, so "regular" `mono_object_new_handle`  has the new better meaning. If this is  confusing, which I agree it could be, perhaps the handle functions should be more mechanically named and resemble the non-handle ones more closely?

I kinda pushed for that but was more than once advised to pick better names, instead of just sticking "handle" somewhere. (exception_new vs. exception_get or something).

Reply:
The renaming is really confusing I agree. Maybe we should revisit the handle allocating names first, line them up with the rawpointer names, as ugly as they are?

Reply:
To be clear, I am not even asserting that I got right. But I did not ignore the comment and I tried to maintain unchanged semantics, except for pointer vs. handle. If I failed, and your question, is perhaps testament that cleaning up the names vs. "just append handle" is not warranted.